include/dr/views/transform.hpp

```diff
@@ -23,12 +20,12 @@ class transform_view : public rng::view_interface<transform_view<V, F>> {
   auto end() const {
     auto segs = segments();
     return normal_distributed_iterator<decltype(segments())>(
-        segs, size_t(segs.size()), 0);
+        segs, std::size_t(segs.size()), 0);
   }
 
   auto segments() const {
     return lib::ranges::segments(base_) |
-           rng::views::transform([=](auto &&segment) {
+           rng::views::transform([this](auto &&segment) {
```

Comment:
I suggest updating the capture to `[*this]` or `[fn_]`.  Technically I think we want to copy `fn_` by value, since that's how it's done everywhere in the public-facing API in the standard, and it's required to be copy constructible.

Reply:
On second thought, we probably really want `=` or `fn_`, since `*this` would also copy the segments by value.  (In theory this should be cheap, but for some of our `segments()` that are vectors with a bunch of stuff, there's no reason to require copying this when we don't need to.)

Reply:
> we probably really want `=` or `fn_`, since `*this`

Neither are accepted by gcc12:
```
 error: implicit capture of 'this' via '[=]' is \
deprecated in C++20 [-Werror=deprecated]
   28 |            rng::views::transform([=](auto &&segment) {
      |                                  ^
```
and
```
/nfs/site/home/rscohn1/local/projects/dds/dr/include/./dr/views/transform.hpp:28:35: error: capture of non-variable 'lib::transform_\
view<V, F>::fn_'
   28 |            rng::views::transform([fn_](auto &&segment) {
      |                                   ^~~
```
This captures the value and compiles without error.
```
auto segments() const {
    auto fn = fn_;
    return lib::ranges::segments(base_) |
           rng::views::transform([fn](auto &&segment) {                                                                                           return segment | rng::views::transform(fn);
           });
  }
```

Reply:
Ah, thanks for catching this... I've run into a couple of cases recently where Clang's front-end lets through things that GCC rejects.  Looks like you found a good fix.